crates/core/src/dbs/processor.rs

```diff
@@ -93,34 +93,44 @@ impl Collected {
 		self,
 		opt: &Options,
 		txn: &Transaction,
+		rid_only: bool,
 	) -> Result<Processed, Error> {
 		match self {
-			Self::Edge(key) => Self::process_edge(opt, txn, key).await,
+			Self::Edge(key) => Self::process_edge(opt, txn, key, rid_only).await,
 			Self::RangeKey(key) => Self::process_range_key(key).await,
 			Self::TableKey(key) => Self::process_table_key(key).await,
 			Self::Relatable {
 				f,
 				v,
 				w,
 				o,
-			} => Self::process_relatable(opt, txn, f, v, w, o).await,
-			Self::Thing(thing) => Self::process_thing(opt, txn, thing).await,
-			Self::Yield(table) => Self::process_yield(opt, txn, table).await,
+			} => Self::process_relatable(opt, txn, f, v, w, o, rid_only).await,
+			Self::Thing(thing) => Self::process_thing(opt, txn, thing, rid_only).await,
+			Self::Yield(table) => Self::process_yield(opt, txn, table, rid_only).await,
 			Self::Value(value) => Ok(Self::process_value(value)),
-			Self::Defer(key) => Self::process_defer(opt, txn, key).await,
-			Self::Mergeable(v, o) => Self::process_mergeable(opt, txn, v, o).await,
+			Self::Defer(key) => Self::process_defer(opt, txn, key, rid_only).await,
+			Self::Mergeable(v, o) => Self::process_mergeable(opt, txn, v, o, rid_only).await,
 			Self::KeyVal(key, val) => Ok(Self::process_key_val(key, val)?),
 			Self::Count(c) => Ok(Self::process_count(c)),
-			Self::IndexItem(i) => Self::process_index_item(opt, txn, i).await,
+			Self::IndexItem(i) => Self::process_index_item(opt, txn, i, rid_only).await,
 			Self::IndexItemKey(i) => Ok(Self::process_index_item_key(i)),
 		}
 	}
 
-	async fn process_edge(opt: &Options, txn: &Transaction, key: Key) -> Result<Processed, Error> {
+	async fn process_edge(
+		opt: &Options,
+		txn: &Transaction,
+		key: Key,
+		rid_only: bool,
+	) -> Result<Processed, Error> {
 		// Parse the data from the store
 		let gra: graph::Graph = graph::Graph::decode(&key)?;
 		// Fetch the data from the store
-		let val = txn.get_record(opt.ns()?, opt.db()?, gra.ft, &gra.fk, None).await?;
+		let val = if rid_only {
+			Arc::new(Value::Null)
```

Comment:
nitpick but usually inside surrealql null means an empty value and none means no value. Should we maybe make these values none instead? Reading through I'm guessing the user will never see this value but just a thought which came up

Reply:
Semantically speaking IMHO null is more appropriate. Because it means there is a record, we are just skipping it. None, means nothing, including record ID.